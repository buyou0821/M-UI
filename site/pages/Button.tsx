import React from 'react';
import { Icon, Button, ConfigProvider } from 'components/index';

export default () => {
  return (
    <ConfigProvider>
      <div className="demo-buttons">
        <h3>Contained Buttons</h3>
        <Button>DEFAULT</Button>
        <Button loading color="primary">
          PRIMARY
        </Button>
        <Button color="secondary" disabled>
          SECONDARY
        </Button>
        <Button href="https://github.com/buyou0821/tyche-ui" target="_blank" color="success">
          Tyche UI
        </Button>
        <Button htmlType="submit" color="warning" ripple={false}>
          WARNING
        </Button>
        <Button color="error">ERROR</Button>
      </div>
      <div className="demo-buttons">
        <h3>Text Buttons</h3>
        <Button shape="text">DEFAULT</Button>
        <Button shape="text" color="primary" center>
          <Icon left material="create" />
          PRIMARY
        </Button>
        <Button shape="text" color="secondary">
          SECONDARY
        </Button>
        <Button shape="text" color="success">
          SUCCESS
        </Button>
        <Button shape="text" color="warning">
          WARNING
        </Button>
        <Button shape="text" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Outlined Buttons</h3>
        <Button shape="outlined">DEFAULT</Button>
        <Button shape="outlined" color="primary">
          PRIMARY
        </Button>
        <Button shape="outlined" color="secondary" disabled>
          SECONDARY
        </Button>
        <Button shape="outlined" color="success">
          SUCCESS
        </Button>
        <Button shape="outlined" color="warning">
          WARNING
        </Button>
        <Button shape="outlined" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Circle/Fab Buttons</h3>
        <Button shape="circle">
          <Icon material="face" />
        </Button>
        <Button shape="circle" loading color="primary">
          <Icon material="remove" />
        </Button>
        <Button shape="circle" color="secondary" disabled>
          <Icon material="create" />
        </Button>
        <Button shape="circle" color="success" fab>
          <Icon material="done" />
        </Button>
        <Button shape="circle" color="warning" fab loading>
          <Icon material="priority_high" />
        </Button>
        <Button shape="circle" color="error" fab disabled>
          <Icon material="clear" />
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Round Buttons</h3>
        <Button shape="round">DEFAULT</Button>
        <Button shape="round" color="primary">
          PRIMARY
        </Button>
        <Button shape="round" color="secondary" disabled>
          SECONDARY
        </Button>
        <Button shape="round" color="success">
          SUCCESS
        </Button>
        <Button shape="round" color="warning">
          WARNING
        </Button>
        <Button shape="round" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Icon Buttons</h3>
        <Button shape="icon">
          <Icon material="face" />
        </Button>
        <Button shape="icon" color="primary">
          <Icon material="remove" />
        </Button>
        <Button shape="icon" color="secondary" disabled>
          <Icon material="create" />
        </Button>
        <Button shape="icon" color="success" size="small">
          <Icon material="done" />
        </Button>
        <Button shape="icon" color="warning">
          <Icon material="priority_high" />
        </Button>
        <Button shape="icon" color="error" size="large">
          <Icon material="clear" />
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Icon Buttons</h3>
        <Button color="primary">
          <Icon left material="delete" />
          删除
        </Button>
        <Button color="secondary" disabled>
          <Icon left material="create" />
          添加
        </Button>
        <Button color="success">
          成功
          <Icon right material="done" />
        </Button>
        <Button color="warning">
          警告
          <Icon right material="warning" />
        </Button>
        <Button color="error">
          <Icon left material="send" />
          SEND
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Size</h3>
        <Button color="primary" size="small">
          <Icon left material="delete" />
          删除
        </Button>
        <Button color="secondary">
          <Icon left material="create" />
          添加
        </Button>
        <Button color="success" size="large">
          成功
          <Icon right material="done" />
        </Button>
        <Button color="error" shape="circle" size="small">
          <Icon material="warning" />
        </Button>
        <Button color="warning" shape="circle">
          <Icon material="warning" />
        </Button>
        <Button color="error" shape="circle" size="large">
          <Icon material="warning" />
        </Button>
      </div>
    </ConfigProvider>
  );
};
