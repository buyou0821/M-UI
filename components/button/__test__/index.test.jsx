import React, { useState } from 'react';
import Button from '..';
import { render, mount } from 'enzyme';

describe('Button', () => {
  it('render correctly', () => {
    const wrapper = render(<Button>Follow</Button>);
    expect(wrapper).toMatchSnapshot();

    // size
    const wrapper1 = render(<Button size="small">Follow</Button>);
    expect(wrapper1).toMatchSnapshot();

    // shape
    const wrapper2 = render(<Button shape="circle">Follow</Button>);
    expect(wrapper2).toMatchSnapshot();

    // color
    const wrapper3 = render(<Button color="success">Follow</Button>);
    expect(wrapper3).toMatchSnapshot();
  });

  it('should support link button', () => {
    const wrapper = mount(
      <Button target="_blank" href="https://github.com/buyou0821/tyche-ui">
        link button
      </Button>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should support ripple attribute', async () => {
    const wrapper = mount(<Button ripple={false}>button</Button>);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should support loading', () => {
    const LoadingButton = () => {
      const [loading, setLoading] = useState(false);
      const toggleLoading = () => {
        setLoading(!loading);
      };
      return (
        <Button onClick={toggleLoading} loading={loading}>
          button
        </Button>
      );
    };
    const wrapper = mount(<LoadingButton />);
    expect(wrapper.find('.ty-btn__loading').length).toBe(0);
    wrapper.simulate('click');
    expect(wrapper.find('.ty-btn__loading').length).toBe(1);
  });
});
