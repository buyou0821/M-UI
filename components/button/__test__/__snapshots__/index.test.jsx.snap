// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly 1`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default"
>
  <span
    class="ty-btn__wrapper"
  >
    Follow
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button render correctly 2`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default ty-btn--small"
>
  <span
    class="ty-btn__wrapper"
  >
    Follow
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button render correctly 3`] = `
<button
  class="ty-ripple ty-btn ty-btn__circle ty-btn__circle--default"
>
  <span
    class="ty-btn__wrapper"
  >
    Follow
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button render correctly 4`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--success"
>
  <span
    class="ty-btn__wrapper"
  >
    Follow
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button should has click ripple effect 1`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default"
>
  <span
    class="ty-btn__wrapper"
  >
    button
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button should not render as link button when href is undefined 1`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default"
>
  <span
    class="ty-btn__wrapper"
  >
    button
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</button>
`;

exports[`Button should support link button 1`] = `
<a
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default"
  href="https://github.com/buyou0821/tyche-ui"
  target="_blank"
>
  <span
    class="ty-btn__wrapper"
  >
    link button
  </span>
  <span
    class="ty-ripple__wrapper"
  />
</a>
`;

exports[`Button should support ripple attribute 1`] = `
<button
  class="ty-ripple ty-btn ty-btn__contained ty-btn__contained--default"
>
  <span
    class="ty-btn__wrapper"
  >
    button
  </span>
</button>
`;
